src/cmd-generate-hashlist

```diff
@@ -0,0 +1,177 @@
+        }]
+    }
+}"""
+
+
+class HashListV1(dict):
+    """
+    Abstraction of a HashList in the version 1 series
+    """
+
+    def __init__(self, release=None, generator=None, timestamp=None):
+        """
+        Initialize HashListV1 instance.
+
+        :param release: Optional release override
+        :type release: str
+        :param generator: Optional generator override
+        :type generator: str
+        :param timestamp: Optional timestamp override
+        :type timestamp: str
+        :raises: KeyError
+        """
+        super()
+        metadata = meta.GenericBuildMeta()
+        if release is None:
+            release = metadata['buildid']
+        self['release'] = release
+
+        self['meta'] = {}
+        if generator is None:
+            generator = 'coreos-assembler-' + metadata['coreos-assembler.container-image-git']['commit']  # noqa
+        self['meta']['generator'] = generator
+        if timestamp is None:
+            timestamp = datetime.datetime.utcnow().isoformat()
+        self['meta']['timestamp'] = timestamp
+        self['meta']['image'] = os.path.sep.join([
+            'builds', metadata['ostree-version'],
+            metadata['coreos-assembler.basearch'],
+            metadata['images']['qemu']['path']])
+
+        self['hashes'] = {}
+
+    def convert_file(self, file_path):
+        """
+        Converts the output from KeyLimes create_list.sh into
+        the HashList hash structure.
+
+        :param file_path: Path to the KeyLime list file
+        :type file_path: str
+        :raises: FileNotFoundError, IsADirectoryError, PermissionError
+        """
+        with open(file_path, 'r') as wp:
+            for line in wp.readlines():
+                sha256, path = line[:-1].split('  ')
+                self['hashes'][path] = [{'sha256': sha256}]
+
+    def write(self, output):
+        """
+        Renders the HashList structure to disk.
+
+        :param output: Where to write the file
+        :type output: str
+        :raises: FileNotFoundError, IsADirectoryError, PermissionError
+        """
+        with open(output, 'w') as out:
+            out.write(json.dumps(self, indent=4))
+
+
+def create_list(hashlist):
+    """
+    Starts up a VM and gets the list created.
+
+    :param hashlist: Iniitialized HashList instance
+    :type hashlist: HashListV1
+    :raises: CalledProcessError, PermissionError
+    """
+    try:
+        os.mkdir('hashmount')
+    except FileExistsError:
+        # It's OK if it already exists
+        pass
+    shutil.copy('/usr/lib/coreos-assembler/generate_list.sh', 'hashmount/generate_list.sh')
+    with tempfile.NamedTemporaryFile(mode='w') as ign_file:
+        ign_file.write(CREATE_HASH_IGN)
+        ign_file.flush()
+        subprocess.check_call([
+            '/usr/bin/kola', 'qemuexec', '--devshell', '-i', ign_file.name,
```

Comment:
`--devshell` was only intended for interactive use.  It might fail if there's not a tty attached (i.e. jenkins build).

Reply:
Good call. I'll test without `--devshell` as it's likely not needed.